pkg/ops/miningManager.go

```diff
@@ -121,26 +169,167 @@ func (mgr *MiningMgr) Start(ctx context.Context) {
 			select {
 			// Boss wants us to quit for the day.
 			case <-mgr.exitCh:
-				//exit
 				input <- nil
 
-			//found a solution
+			// Found a solution.
 			case result := <-output:
+				if mgr.profitThreshold > 0 {
+					isProftable, err := mgr.isProfitable()
+					if err != nil {
+						mgr.log.Warn("error calculating the solution submition profitability so skiping")
+						return
+					}
+					if !isProftable {
+						mgr.log.Warn("solution submition not profitable so skiping")
+						return
+					}
+				}
 				if result == nil {
 					mgr.Running = false
 					return
 				}
 				mgr.solution = result
-				goodSubmit := mgr.solHandler.Submit(ctx, mgr.solution)
-				if goodSubmit {
-					fmt.Println("solution submitted") //mgr.solution = nil
+				tx, err := mgr.solHandler.Submit(ctx, mgr.solution)
```

Comment:
move profit check here (below setting mgr.solution so it doesn't have to resolve the solution)

Reply:
I updated the logic please double check now. 
With the new logic it just blocks until a transaction is profitable and only then it submits.

One problem I see with this approach is that by the time a transaction is profitable the challenge might have changed and the solution would be invalid, but fixing this would require some greater refactorings and I would say to leave it for another PR. I also added a new issue to simplify the logic in the manager here as I think it can be much more simple than now.
tracking issue https://github.com/tellor-io/TellorMiner/issues/232

Reply:
Completely refactored the code flow and it is not blocked and I think it is much easier to follow.